src/System.Windows.Forms/src/System/Windows/Forms/Nrbf/WinFormsSerializationRecordExtensions.cs

```diff
@@ -53,6 +56,35 @@ public static bool TryGetBitmap(this SerializationRecord record, out object? bit
         return true;
     }
 
+    /// <summary>
+    ///  Tries to deserialize this object if it was serialized as JSON.
+    /// </summary>
+    public static bool TryGetObjectFromJson(this SerializationRecord record, out object? @object)
+    {
+        @object = null;
+
+        if (record.TypeName.AssemblyName?.FullName != IJsonData.CustomAssemblyName)
+        {
+            // The data was not serialized as JSON.
+            return false;
+        }
+
+        if (record is not ClassRecord types
+            || types.GetRawValue("<JsonBytes>k__BackingField") is not SZArrayRecord<byte> byteData
+            || !TypeName.TryParse(types.TypeName.FullName, out TypeName? result)
+            || result.GetGenericArguments().FirstOrDefault() is not { } genericTypeName
```

Comment:
```suggestion
            || result.GetGenericArguments().Single() is not { } genericTypeName
```
THere should be exactly 1 generic parameter

Reply:
Oh thanks! I didn't know about this method 😄